src/main/java/bio/terra/service/dataset/DatasetTableDao.java

```diff
@@ -58,6 +61,28 @@ public void createTables(UUID parentId, List<DatasetTable> tableList) {
             params.addValue("raw_table_name", table.getRawTableName());
             params.addValue("soft_delete_table_name", table.getSoftDeleteTableName());
 
+            PartitionMode mode = table.getPartitionMode();
+            params.addValue("partition_mode", mode.getName());
+
+            if (mode instanceof DatePartitionMode) {
```

Comment:
I was thinking things like this could be in the subclasses. Could have the base class implement the null return... but it would require an object for the no mode case.

Don't feel strongly about it.

Reply:
This is where I could see using a `json` column to store the partitioning options could make sense, so there's just one method to override that returns whatever-it-needs-to, with a complementary `static` builder to reconstruct the appropriate mode based on the JSON.

Reply:
Actually, that might help with the DAO/PDAO blurring too... I'm going to play with it a bit before merging this.